packages/cli/src/index.ts

```diff
@@ -149,22 +142,28 @@ const main = async () => {
   }
 
   // Print update information, if available
-  if (notifier.update && notifier.update.latest !== pkg.version && isTTY) {
-    const { latest } = notifier.update;
-    console.log(
-      info(
-        `${chalk.black.bgCyan('UPDATE AVAILABLE')} ` +
-          `Run ${cmd(
-            await getUpdateCommand()
-          )} to install ${getTitleName()} CLI ${latest}`
-      )
-    );
+  if (isTTY && !process.env.NO_UPDATE_NOTIFIER) {
+    // Checks for available update and returns an instance
+    const latest = await updateNotifier({
```

Comment:
@cb1kenobi The fact that we are awaiting here makes me think this is not running in the background. It should read the file and continue but only write the file in the background.

Reply:
The `await` has nothing to do with whether it's running in the background or not. `updateNotifier()` performs async logic prior to spawning the worker process and we need to block in case an error occurs (which is unlikely).

Another reason we `await` was added was because I added a `wait` option which will block until the update check completes and is useful for debugging.

Blocking while it quickly tries to read the cache file is a relatively cheap operation compared to always spawning the process to perform the check.